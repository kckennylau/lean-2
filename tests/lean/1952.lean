structure foo :=
(fn    : nat → nat)
(fn_ax : ∀ a : nat, fn a = a)

/-
set_option pp.all true
set_option pp.universes false                 -- universes are probably irrelevant
set_option pp.purify_metavars false
set_option trace.type_context.is_def_eq true
set_option trace.type_context.is_def_eq_detail true
-/

def bla : foo := { fn_ax := λ x, rfl }


instance foo2 (α : Type) : preorder α := { le_trans := λ x y z, rfl }
