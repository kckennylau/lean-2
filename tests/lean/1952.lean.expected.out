1952.lean:13:17: error: invalid structure value { ... }, field 'fn' was not provided
1952.lean:13:28: error: type mismatch at field 'fn_ax'
  λ (x : ℕ), rfl
has type
  ∀ (x : ℕ), ?m_2[x] = ?m_2[x]
but is expected to have type
  ∀ (a : ℕ), ⁇ a = a
1952.lean:16:41: error: invalid structure value { ... }, field 'le' was not provided
1952.lean:16:41: error: invalid structure value { ... }, field 'le_refl' was not provided
1952.lean:16:55: error: type mismatch at field 'le_trans'
  λ (x y z : α), rfl
has type
  ∀ (x y z : α), ?m_2[x, y, z] = ?m_2[x, y, z]
but is expected to have type
  ∀ (a b c : α), a ≤ b → b ≤ c → a ≤ c
